cmake/BLTMacros.cmake

```diff
@@ -43,6 +43,52 @@
 
 include(${BLT_ROOT_DIR}/cmake/BLTPrivateMacros.cmake)
 
+##------------------------------------------------------------------------------
+## blt_list_append( TO <list> ELEMENTS [ <element>...] IF <bool> )
+##
+## Appends elements to a list if the specified bool evaluates to true.
+##
+## This macro is essentially a wrapper around CMake's `list(APPEND ...)`
+## command which allows inlining a conditional check within the same call
+## for clarity and convenience.
+##
+## This macro requires specifying:
+##   (1) The target list to append to by passing TO <list>
+##   (2) A condition to check by passing IF <bool>
```

Comment:
Please make a note that the IF argument cannot handle anything other than a single variable.  For example, the following won't work:

blt_list_append( TO mylist ELEMENTS C
                            IF NOT ${ENABLE_C} )

Reply:
Yes, good point. I'll add a comment. 

As a side note, we might be able to make use of the boolean generator expression $<BOOL: NOT ${ENABLE_C}> to inline evaluate a boolean expression and pass the result to the macro. I'll give this a shot and add comments accordingly.

Reply:
I like that idea

Reply:
Another option is to add an IF_NOT. Though this is a pretty slippery slope. IF_AND, IF_OR.. where does it stop.

Reply:
Yeah, it's a rabbit hole. Unfortunately, the `$<BOOL: ...>` doesn't work in this context either. I added a clarification to the documentation for the macro.